test/mailers/subscription_mailer_test.rb

```diff
@@ -81,5 +94,17 @@ def test_omits_non_expiring_subscription
       assert_equal rows.length, 2
       assert Katello::Pool.readable.size > 1
     end
+
+    def test_days_from_now_mail_query
+      pool_expiring_soon.save
```

Comment:
I think if you do `FactoryBot.create` vs `FactoryBot.build` you don't need the `save` calls~

Reply:
@jturel For this test I wanted to ensure that both records were present when `deliver` is run.  Because the `let`s are lazily evaluated, they might not get created until the asserts, which would mean that this test isn't really testing anything.  (I could probably change it for the other tests though)

Reply:
ah ok. if you use create vs build then you can call just `pool_expiting_soon` here

Reply:
@jturel Thanks, updated!